test/envtest/metrics_envtest_test.go

```diff
@@ -27,140 +27,84 @@ func TestMetricsAreServed(t *testing.T) {
+		metrics.MetricNameFallbackConfigPushSuccessTime,
+		metrics.MetricNameFallbackConfigPushDuration,
+		metrics.MetricNameFallbackConfigPushBrokenResources,
+		metrics.MetricNameProcessedConfigSnapshotCacheHit,
+		metrics.MetricNameProcessedConfigSnapshotCacheMiss,
+	}
+
 	scheme := Scheme(t, WithKong)
 	envcfg := Setup(t, scheme)
 
-	testCases := []struct {
-		name                         string
-		withPushError                bool
-		fallbackConfigurationEnabled bool
-		expectedMetrics              []string
-		skippedMessage               string
-	}{
-		{
-			name:                         "with push error and FallbackConfiguration enabled",
-			skippedMessage:               "flaky, see https://github.com/Kong/kubernetes-ingress-controller/issues/6125",
-			withPushError:                true,
-			fallbackConfigurationEnabled: true,
-			expectedMetrics: []string{
-				metrics.MetricNameConfigPushCount,
-				metrics.MetricNameConfigPushBrokenResources,
-				metrics.MetricNameTranslationCount,
-				metrics.MetricNameTranslationDuration,
-				metrics.MetricNameConfigPushSize,
-				metrics.MetricNameTranslationBrokenResources,
-				metrics.MetricNameConfigPushDuration,
-
-				metrics.MetricNameFallbackTranslationBrokenResources,
-				metrics.MetricNameFallbackTranslationCount,
-				metrics.MetricNameFallbackTranslationDuration,
-				metrics.MetricNameFallbackConfigPushSize,
-				metrics.MetricNameFallbackConfigPushCount,
-				metrics.MetricNameFallbackConfigPushSuccessTime,
-				metrics.MetricNameFallbackConfigPushDuration,
-				metrics.MetricNameFallbackConfigPushBrokenResources,
-				metrics.MetricNameProcessedConfigSnapshotCacheHit,
-				metrics.MetricNameProcessedConfigSnapshotCacheMiss,
-			},
-		},
-		{
-			name:          "without push error",
-			withPushError: false,
-			expectedMetrics: []string{
-				metrics.MetricNameConfigPushCount,
-				metrics.MetricNameConfigPushBrokenResources,
-				metrics.MetricNameTranslationCount,
-				metrics.MetricNameTranslationDuration,
-				metrics.MetricNameConfigPushSize,
-				metrics.MetricNameTranslationBrokenResources,
-				metrics.MetricNameConfigPushDuration,
-				metrics.MetricNameConfigPushSuccessTime,
-			},
-		},
+	// We're going to make the first request to the admin API fail, so that the manager falls back to the last valid
+	// configuration and we can test the fallback metrics.
+	adminAPIOpts := []mocks.AdminAPIHandlerOpt{
+		mocks.WithConfigPostError([]byte(`{"flattened_errors": [{"errors": [{"messages": ["broken object"]}], "entity_tags": ["k8s-name:test-service","k8s-namespace:default","k8s-kind:Service","k8s-uid:a3b8afcc-9f19-42e4-aa8f-5866168c2ad3","k8s-group:","k8s-version:v1"]}]}`)),
+		mocks.WithConfigPostErrorOnlyOnFirstRequest(),
 	}
+	addr := fmt.Sprintf("localhost:%d", helpers.GetFreePort(t))
+	_ = RunManager(t.Context(), t, envcfg,
+		AdminAPIOptFns(adminAPIOpts...),
+		func(cfg *managercfg.Config) {
+			cfg.FeatureGates[managercfg.FallbackConfigurationFeature] = true
+		},
```

Comment:
We could add an opt func to set feature gates which could be used like: `WithFeatureGate(managercfg.FallbackConfigurationFeature, true)`. WDYT?

Reply:
Good idea: [fe7dda1](https://github.com/Kong/kubernetes-ingress-controller/pull/7310/commits/fe7dda1a90e4789cd1a5bf711fccf03968fa2155)